lnwallet/commitment.go

```diff
@@ -244,11 +241,12 @@ func CommitScriptToSelf(chanType channeldb.ChannelType, initiator bool,
 		}
 
 		toLocalPkScript, err := input.PayToTaprootScript(
-			toLocalOutputKey,
+			toLocalScriptTree.TaprootKey,
 		)
 
 		return &ScriptInfo{
-			PkScript: toLocalPkScript,
+			WitnessScript: toLocalScriptTree.SettleLeaf.Script,
```

Comment:
Not sure what you mean `CommitScriptToSelf` is always called contextually: the `selfKey` here will always be the key of the party that has the longer delay output. See `CreateCommitTx`.

Reply:
in `NewBreachRetribution` we need the witness script to be the revocation script, not the success script. This sets it to the revocation script.

Reply:
Ah I see what you mean now, yeah you're right! Pre taproot it doesn't matter as the entire script stays the same, with only the revocation path taken. In this case, we need to worry about the revocation path as well. Thankfully the returned script tree has the revocation leaf contained, so we can update the call site to use that instead.

Reply:
could still add a `if initiator: witness=successLeaf; else witness=revocationLeaf` block here so that the witness stays correct :) else a caller explicitly needs to know when they should reconstruct it from the script tree vs just use the witness as is